File: truckdoc-client-m2/application/src/main/java/com/sanda/truckdoc/client/data/model/route/DbRoutePath.java
package com.sanda.truckdoc.client.data.model.route;

import com.j256.ormlite.dao.ForeignCollection;
import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.field.ForeignCollectionField;
import com.j256.ormlite.table.DatabaseTable;
import com.sanda.truckdoc.client.api.v3.sync.routing.model.RoutePath;

import org.jetbrains.annotations.NotNull;

/**
 *
 */
@DatabaseTable(tableName = "route_pathes")
public class DbRoutePath {


    @DatabaseField(generatedId = true)
    private long id;
    @DatabaseField
    private String name;
    @DatabaseField
    private double serverRouteId;
    @ForeignCollectionField(eager = true)
    private ForeignCollection<DbRoutePoint> points;

    @DatabaseField(canBeNull = false, foreign = true)
    private DbRouteAssignment routeAssignment;

    public DbRoutePath() {
    }

    public DbRoutePath(@NotNull RoutePath routePath, @NotNull Long routeId) {
        this.name = routePath.getName();
        this.serverRouteId = routeId;
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getServerRouteId() {
        return serverRouteId;
    }

    public void setServerRouteId(double serverRouteId) {
        this.serverRouteId = serverRouteId;
    }

    public ForeignCollection<DbRoutePoint> getPoints() {
        return points;
    }

    public void setPoints(ForeignCollection<DbRoutePoint> points) {
        this.points = points;
    }

    public DbRouteAssignment getRouteAssignment() {
        return routeAssignment;
    }

    public void setRouteAssignment(DbRouteAssignment routeAssignment) {
        this.routeAssignment = routeAssignment;
    }

    @Override
    public String toString() {
        final StringBuffer sb = new StringBuffer("DbRoutePath{");
        sb.append("id=").append(id);
        sb.append(", name=").append(name);
        sb.append(", serverRouteId=").append(serverRouteId);
        sb.append('}');
        return sb.toString();
    }
}

File: truckdoc-client-m2/data/src/main/java/com/sanda/truckdoc/client/data/model/route/DbRoutePath.kt
package com.sanda.truckdoc.client.data.model.route

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import com.sanda.truckdoc.client.api.v3.sync.routing.model.RoutePath

@Entity(tableName = "route_paths", foreignKeys = [
    ForeignKey(entity = DbRouteAssignment::class, parentColumns = ["id"], childColumns = ["routeAssignmentId"], onDelete = ForeignKey.CASCADE)
])
data class DbRoutePath constructor(
        @PrimaryKey(autoGenerate = true)
        val id: Long = 0,
        var name: String,
        var serverRouteId: Double,
        //@DatabaseField(canBeNull = false, foreign = true)
        var routeAssignmentId: Long = 0
) {
    constructor(routePath: RoutePath, routeId: Long) : this(name = routePath.name,
            serverRouteId = routeId.toDouble())
}
